### ISSUE TITLE
Failures in download and repair: write at specified offset exceeds buffer size

### ISSUE THREAD

**Issue by mtlynch** [NebulousLabs/Sia#2146]
I'm testing from source (at bf41b5d5781844d69e94b91da76d1dd19b4ff615) in a Docker container and I'm seeing a few instances of this in the `renter.log`:

```
2017/07/16 21:29:31.540376 download.go:593: Download failed - could not recover a chunk: unable to write to download destination: write at specified offset exceeds buffer size
2017/07/16 21:29:31.540441 repair.go:295: Unable to repair chunk: unable to get repair chunk:: unable to write to download destination: write at specified offset exceeds buffer size
```

I can download my files, but my uploads have been stuck at the same percentages for days:

```
$ docker exec -it compose_sia_1 siac renter -v
Renter info:
        Storage Spending:  276 mS
        Upload Spending:   16.7 mS
        Download Spending: 54.73 mS
        Unspent Funds:     1.273 KS
        Total Allocated:   1.273 KS

Tracking 5 files:
File size  Available  Progress  Redundancy  Renewing  Sia path
242.01 MB  Yes          99.44%        2.90  Yes       [redacted]
    804 B  Yes          93.33%        2.80  Yes       [redacted]
104.86 MB  Yes          92.22%        2.40  Yes       [redacted]
  58.6 KB  Yes          93.33%        2.80  Yes       [redacted]
  59.8 KB  Yes          73.33%        2.20  Yes       [redacted]
```

Dockerfile:

```
FROM golang

RUN go get -u github.com/NebulousLabs/Sia/...

RUN apt-get update && apt-get install -y socat

# Make the Sia ports available to the Docker container's host.
EXPOSE 8000 9981 9982

# Configure the Sia daemon to run when the container starts.
# Specify the Sia directory as /mnt/sia so that you can view these files outside
# of Docker.
ENTRYPOINT socat tcp-listen:8000,reuseaddr,fork tcp:localhost:9980 & $GOPATH/bin/siad --modules gctwr --sia-directory /mnt/sia-data
```

**Comment by Cinerar** [comment]
I believe FROM golang in dockerfile is pulling latest, not sure it is best practice, coz it can be some golang bugs.

**Comment by lukechampine** [comment]
this bug is fixed in v1.3.0
